cpp/src/arrow/compute/api_vector.h

```diff
@@ -601,7 +596,7 @@ ARROW_EXPORT
 Result<Datum> CumulativeSum(
     const Datum& values,
     const CumulativeSumOptions& options = CumulativeSumOptions::Defaults(),
-    ExecContext* ctx = NULLPTR);
+    bool check_overflow = false, ExecContext* ctx = NULLPTR);
```

Comment:
Can you document the  additional param in the docstring above?
Also, can you add a small test that the param is taken into account?

Reply:
There were no doc for CumulativeSum above... But I added one anyways.
Also added a small test.